### PR TITLE
fix(scraper): coerce list-valued expert counts; a failed estimate cannot abort a scrape

### PR DESCRIPTION
**Pull request by AlexsJones**
## Problem

The weekly run on 2dd0628 ([35441264888](https://github.com/AlexsJones/llmfit/actions/runs/35441264888)) crashed two minutes in, at discover ~1465/6241:

```
File "scripts/scrape_hf_models.py", line 999, in estimate_params_from_arch
    ffn_total = expert_ffn + shared_ffn + router
TypeError: can only concatenate list (not "int") to list
```

`baidu/ERNIE-4.5-VL-28B-A3B-PT` declares `moe_num_experts` as a per-modality list (`[64, 64]`). The shared expert-key helper I added in #1055's review follow-up started reading that key. `list * int` is silently a list, so nothing failed until the addition one line later. The previous run missed it only because the quantized-only gate (removed in #1063) skipped the estimator for that repo.

Left alone, the list would also have reached the catalog's integer `num_experts` field through `detect_moe` and failed the embedded Rust parse, the same class of failure as the `-1` sentinel in #1055.

## Change

- `_config_num_experts` / `_config_active_experts` return the first **positive int**, taking the head of a list (as the file already does for `moe_intermediate_size`). Empty lists, zeros and bools are not counts.
- `correct_packed_param_count` catches an estimator failure, logs it, and keeps the reported count. One odd `config.json` must not discard a 6,000-model scrape; the estimate is an optional improvement, not load-bearing.

## Validation

```
python3 scripts/test_preserve_catalog_metadata.py    37 passed (2 new)
```

Reproduced locally against the 18 repos in flight when the run died plus the ERNIE family (32 repos): the VL repo's list-valued keys are confirmed, 0 crashes, and no list or dict reaches the `detect_moe` / `extract_arch_metadata` output.

### CODE REVIEW

File: scripts/scrape_hf_models.py [hunk 1]
```diff
@@ -1526,7 +1543,14 @@ def correct_packed_param_count(repo_id: str, total_params: int,
             return declared
         return total_params
 
-    arch_params = estimate_params_from_arch(config)
+    try:
+        arch_params = estimate_params_from_arch(config)
+    except (TypeError, ValueError, ZeroDivisionError) as e:
```
[the remark this answers is not in the document]
Comment (AlexsJones) [reply scripts/scrape_hf_models.py]: Fixed in 93af85e, in two layers: the estimator guard now also catches `AttributeError`/`KeyError`/`IndexError` (test with `text_config: null`), and discovery has the per-model boundary you noted was missing, so any one repo is logged and skipped while its retained entry stands. Skips are counted and more than 25 fails the run, so a real builder bug cannot hide as a thinned catalog.
